backend/frontend_integration.py

```diff
@@ -0,0 +1,165 @@
+import os
+import sys
+import time
+from flask import jsonify, Blueprint, request, send_from_directory
+from http import HTTPStatus
+from sqlalchemy.exc import OperationalError
+from flask_sqlalchemy import SQLAlchemy
+
+from database import db, Recordings, NewSensor, FakeUser
+# This is hack, but it's the simplest way to get things to work without changing things - Daniel
+sys.path.append(os.path.join(os.path.dirname(__file__), 'data_analysis'))
+from data_analysis.metric_analysis import AnalysisController
+from data_analysis.data_types import Recording
+
+
+STATIC_FOLDER = "static"
+endpoints = Blueprint('endpoints', __name__, template_folder=STATIC_FOLDER)
+# piezo_model = Recording.from_file('ctrl_model.yaml')
+
+@endpoints.route('/api/list_recordings/<int:sensor_id>')
+def get_recording_ids(sensor_id: int):
+    try:
+        recordings = db.session.query(Recordings).filter(Recordings.sensorid == sensor_id).all()
+        recording_ids = [recording._id for recording in recordings]
+        response = jsonify(recording_ids)
+        response.headers.add('Access-Control-Allow-Origin', '*')
+        return response
+    except Exception as e:
+        return jsonify(error=f"Error processing request: {str(e)}"), HTTPStatus.BAD_REQUEST
+
+
+@endpoints.route('/recording/<int:recording_id>')
+def plot_recording(recording_id: int):
+    recording = db.session.query(Recordings).filter(Recordings._id == recording_id).first()
+    sensor = db.session.query(NewSensor).filter(NewSensor._id == recording.sensorid).first()
+    rec = Recording.from_real_data(sensor.fs, recording.ts_data)
+    return rec.plot(show=False).to_html()
+
+
+@endpoints.route('/api/get_metrics/<email>')
+def get_metrics(email: str):
+    try:
+        # fs from NewSensor
+        # ts_data, date, sensorid from recordings
+        print(email)
+        user = db.session.query(FakeUser).filter(FakeUser.email == email).first()
+        sensor = db.session.query(FakeUser).join(NewSensor, NewSensor.userid == FakeUser._id).filter(NewSensor.userid == user._id).first() # sampling
```

Comment:
Since sensorid is nullable, how are we gonna get any recordings? Maybe have a default sensor id that we query if it's null?